### PR TITLE
add protobuf-c support for ping/callreq/result message

### PR DESCRIPTION
**Pull request by wengyanqing**
## Description
Use protobuf-c to refactor communication framework. Currently PING/CALLREQ/RESULT messages have been implemented. And other messages will be implemented later. Also current implementation is based on pyclient and rclient will be changed according later.

## Tests
It's just refactor communication framework, all regression test will be reused.

## Additional Notes
This PR is based on 6X_STABLE for review.

## User Story or Issue
https://www.pivotaltracker.com/story/show/168366805

### CODE REVIEW

**Comment by yydzero** [comment]
Shall we add some tests?

**Comment by HaozhouWang** [comment]
Hi Yandong,

Thank you for your review, this PR is for reviewing only and still in WIP. We will not merge this PR into 6X_STABLE and will close it once we get a complete solution.

**Comment by gfphoenix78** [comment]
@wengyanqing I have 2 questions.
1. What is the version of protobuf we use?
2. How do we distribute/install protobuf?

File: src/plcontainer.c [hunk 1]
```diff
@@ -338,7 +339,7 @@ static plcProcResult *plcontainer_get_result(FunctionCallInfo fcinfo,
 		if (conn != NULL) {
 			int res;
 
-			res = plcontainer_channel_send(conn, (plcMessage *) req);
+			res = plcontainer_channel_send(conn, (plcMessage *) plcMsgCallreqToProto(req));
```
Comment (gfphoenix78) [inline src/plcontainer.c]: Memory leaks.
`plcMsgCallreqToProto(req)` returns a memory block allocated via palloc, and there is no memory free.
The same to other use cases.

**Comment by gfphoenix78** [comment]
This PR is no longer used.

**Pull request closed without merging**